### PR TITLE
OPSEXP-933: Add node selector for deployments on ACS Helm Chart

### PR DESCRIPTION
**Pull request by Mieszko96**
Ref. OPSEXP-933

### CODE REVIEW

**Review by alxgomz — CHANGES_REQUESTED** [review]
As I read it. Current implem would make ALL pods scheduled to the same kubernetes nodes. A nodeSelector should be "set-able" for each pod without bringing all the others pods with him.

**Comment by Mieszko96** [comment]
I don't think if it is necessary for every pod to have their own value. But if you think we want this i can add seperate value for every pod e.g values.imagemagick.nodeSelector

**Comment by mteodori** [comment]
> I don't think if it is necessary for every pod to have their own value. But if you think we want this i can add seperate value for every pod e.g values.imagemagick.nodeSelector

for maximum flexibility I would go as suggested by @alxgomz to have a separate value per each component like the ingress one and a root value for components that are single service in the charts subdir

**Comment by alxgomz** [comment]
> I don't think if it is necessary for every pod to have their own value. But if you think we want this i can add seperate value for every pod e.g values.imagemagick.nodeSelector

You're right that doesn't make sense for every pod. But just because you set a node selector for one, the other shouldn't have it. That's more what I meant.
Nodeselector make sense mainly for:
transformers (to isolate cpu load)
activemq & search & postgres & repo (for pv in simplistic config)
other thatn that... that would be just for more flexibility

**Comment by Mieszko96** [comment]
Tested every deployment except deployment-ai-transformer.yaml
With extra values.

activemq:
  nodeSelector: {"kubernetes.io/hostname": multinode-demo-m02}
imagemagick:
  nodeSelector: {"kubernetes.io/hostname": multinode-demo-m02}
libreoffice:
  nodeSelector: {"kubernetes.io/hostname": multinode-demo-m02}
pdfrenderer:
  nodeSelector: {"kubernetes.io/hostname": multinode-demo-m02}
repository:
  nodeSelector: {"kubernetes.io/hostname": multinode-demo-m02}
share:
  nodeSelector: {"kubernetes.io/hostname": multinode-demo-m02}
tika:
  nodeSelector: {"kubernetes.io/hostname": multinode-demo-m02}
transformmisc:
  nodeSelector: {"kubernetes.io/hostname": multinode-demo-m02}
alfresco-digital-workspace:
  nodeSelector: {"kubernetes.io/hostname": multinode-demo-m02}
filestore:
  nodeSelector: {"kubernetes.io/hostname": multinode-demo-m02}
transformrouter:
  nodeSelector: {"kubernetes.io/hostname": multinode-demo-m02}
alfresco-search:
  nodeSelector: {"kubernetes.io/hostname": multinode-demo-m02}
  alfresco-insight-zeppelin:
    nodeSelector: {"kubernetes.io/hostname": multinode-demo-m02}
    insightzeppelin:
      enabled: True
alfresco-sync-service:
  nodeSelector: {"kubernetes.io/hostname": multinode-demo-m02}
postgresql:
  primary:
    nodeSelector: {"kubernetes.io/hostname": multinode-demo-m02}
postgresql-syncservice:
  primary:
    nodeSelector: {"kubernetes.io/hostname": multinode-demo-m02}

ooi:
  enabled: True
  nodeSelector: {"kubernetes.io/hostname": multinode-demo-m02}

msTeams:
  enabled: True
msTeamsService:
  nodeSelector: {"kubernetes.io/hostname": multinode-demo-m02}

File: helm/alfresco-content-services/README.md [hunk 1]
```diff
@@ -151,7 +157,7 @@ Hence, setting up explicit Container memory and then assigning a percentage of i
 | msTeamsService.teams.chat.filenameEnabled | bool | `true` |  |
 | msTeamsService.teams.chat.imageEnabled | bool | `true` |  |
 | msTeamsService.teams.chat.metadataEnabled | bool | `true` |  |
-| ooi | object | `{"enabled":false}` | Choose if you want Office Online Integration capabilities (Alfresco Collaboration Connector for Microsoft 365) |
+| ooi | object | `{"enabled":false,"nodeSelector":{}}` | Choose if you want Office Online Integration capabilities (Alfresco Collaboration Connector for Microsoft 365) |
```
Comment (gionn) [inline helm/alfresco-content-services/README.md:160]: this change looks unintended

File: helm/alfresco-content-services/README.md [hunk 2]
```diff
@@ -35,6 +35,7 @@ Hence, setting up explicit Container memory and then assigning a percentage of i
 | Key | Type | Default | Description |
 |-----|------|---------|-------------|
 | activemq.enabled | bool | `true` |  |
+| activemq.nodeSelector | object | `{}` |  |
```
Comment (gionn) [inline helm/alfresco-content-services/README.md:38]: add a brief description to make clear what is this about?

File: helm/alfresco-content-services/README.md (hunk 1 above)
Comment (Mieszko96) [reply helm/alfresco-content-services/README.md]: Moved this variable from ooi to ooiService

File: helm/alfresco-content-services/README.md (hunk 2 above)
Comment (Mieszko96) [reply helm/alfresco-content-services/README.md]: I added, but only to activemq.NodeSelector, cuz there is a need to change it in values.yml and i don't wanna spam 15times same sentence there :)

**Review by gionn — APPROVED** [review]

**Comment by scila1996** [comment]
Please add their parameter

`NodeAffinity`
`PodAffinity`
`Pod Tolerence`

**Comment by gionn** [comment]
@scila1996 can you please open a new issue, describing why do you need them (which is your use case)? Thanks.

**Review by alxgomz — APPROVED** [review]
LGTM

**Pull request merged**